src/components/Picklist/__test__/picklist.spec.js

```diff
@@ -48,7 +58,497 @@ describe('<Picklist />', () => {
             .instance()
             .activeChildren.findIndex(child => child.name === 'option1');
         expect(optionUnregisterChildFn).toHaveBeenCalled();
-        expect(component.instance().activeChildren.length).toEqual(2);
-        expect(itemIndex).toEqual(-1);
+        expect(component.instance().activeChildren.length).toBe(2);
+        expect(itemIndex).toBe(-1);
+    });
+    it('should open menu when click the picklist input', () => {
+        const component = mount(
+            <Picklist label="Picklist">
+                <PicklistOption label="Option 1" name="option1" />
+                <PicklistOption label="Option 2" name="option2" />
+                <PicklistOption label="Option 3" name="option3" />
+            </Picklist>,
+        );
+        jest.runAllTimers();
+        expect(component.state().isOpen).toBe(false);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist',
+        );
+        component.instance().menuRef.current = Object.assign(
+            {},
+            component.instance().menuRef.current,
+            {
+                clientHeight: 225,
+                scrollTop: 0,
+                scrollHeight: 285,
+                scrollTo: jest.fn(() => {}),
+            },
+        );
+        component.find('input').simulate('click');
+        expect(component.state().isOpen).toBe(true);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist rainbow-picklist--open',
+        );
+    });
+    it('should open menu when picklist is focused and press UP navigation key', () => {
+        const component = mount(
+            <Picklist label="Picklist">
+                <PicklistOption label="Option 1" name="option1" />
+                <PicklistOption label="Option 2" name="option2" />
+                <PicklistOption label="Option 3" name="option3" />
+            </Picklist>,
+        );
+        jest.runAllTimers();
+
+        const input = component.find('input');
+        input.simulate('focus');
+
+        expect(component.state().isOpen).toBe(false);
+        expect(component.find("div[role='presentation']").prop('className')).toBe(
+            'rainbow-picklist',
+        );
+        component.instance().menuRef.current = Object.assign(
+            {},
+            component.instance().menuRef.current,
+            {
+                clientHeight: 225,
+                scrollTop: 0,
+                scrollHeight: 285,
+                scrollTo: jest.fn(() => {}),
+            },
+        );
```

Comment:
DRY (Don't Repeat Yourself):
you can set component.instance().menuRef.current to an object only with what is needed and store in a variable outside like:
```
const menuRef = {
                clientHeight: 225,
                scrollTop: 0,
                scrollHeight: 285,
                scrollTo: jest.fn(() => {}),
            };
```
then in the test:
```component.instance().menuRef.current = menuRef```
and when you need to use other value:
```
component.instance().menuRef.current = {
    ...menuRef,
    scrollHeight: 400,
}
```